apps/lastfm/last_fm.star

```diff
@@ -0,0 +1,130 @@
+"""
+Applet: Last FM
+Summary: Show Last.fm history
+Description: Show title, artist and album art from most recently scrobbled song in your Last.fm history.
+Author: Chuck
+"""
+
+load("render.star", "render")
+load("schema.star", "schema")
+load("http.star", "http")
+load("encoding/base64.star", "base64")
+load("cache.star", "cache")
+load("time.star", "time")
+load("secret.star", "secret")
+
+def main(config):
+    userName = config.get("lastFmUser") or "badUser"
+    api_key = config.get("lastApiKey") or "badKey"
+    clockShown = config.get("showClock") or True
+
+    # handle missing config data
+    if (userName == "badUser"):
+        print("bad user")
+        return render.Root(
+            child = render.WrappedText(
+                content = "Last.fm Username missing in Tidbyt config.",
+                color = "#FF0000",
+                font = "tom-thumb",
+            ),
+        )
+    if (api_key == "badKey"):
+        print("bad key")
+        return render.Root(
+            child = render.WrappedText(
+                content = "Last.fm API key missing in Tidbyt config.",
+                color = "#FF0000",
+                font = "tom-thumb",
+            ),
+        )
+
+    lastFmUrl = "http://ws.audioscrobbler.com/2.0/?method=user.getrecenttracks&user=" + userName + "&api_key=" + api_key + "&format=json"
+
+    rep = http.get(lastFmUrl)
```

Comment:
Since users have to provide their own API key, this isn't a huge issue, but it could still be a good idea to add some caching for this API call.

Reply:
Thanks for the quick review. 
I updated the schema with default "demo" values for username/key. Instead of caching, I'm still displaying the error text when username/key are blank, but I've added "demo" directions in the appropriate error messages. When default "DemoUser" or "DemoKey" values are used, I now display a mocked up an album cover for demo purpose as you suggested...it's a hefty base64 string, not sure if there's a better option without calling out. Re-submitting for review now. Thanks again.